frontend/src/api/mutations/BuyLiquidity.ts

```diff
@@ -0,0 +1,133 @@
+import { useActiveWalletType, useCosmWasmSigningClient } from "graz"
+import { useMutation, useQueryClient } from "@tanstack/react-query"
+import type { SigningCosmWasmClient } from "@cosmjs/cosmwasm-stargate"
+
+import { useCurrentAccount } from "@config/chain"
+import { useNotifications } from "@config/notifications"
+import { querierAwaitCacheAnd, querierBroadcastAndWait } from "@api/querier"
+import { MARKET_KEYS, type MarketId, type OutcomeId } from "@api/queries/Market"
+import { POSITIONS_KEYS } from "@api/queries/Positions"
+import { BALANCES_KEYS } from "@api/queries/Balances"
+import { trackBuyLiquidity } from "@utils/analytics"
+import type { Coins } from "@utils/coins"
+import { AppError, errorsMiddleware } from "@utils/errors"
+
+export const LIQUIDITY_PORTION = "1"
+
+interface BuyLiquidityRequest {
```

Comment:
You should use the `Provide` execute message instead, it's designed for providing liquidity without buying shares.

Reply:
Thanks for catching that, I hadn't realized